config.json

```diff
@@ -2661,7 +2661,22 @@
           "bitstrings"
         ],
         "difficulty": 6
-      }
+       },
+       {
+        "slug": "resistor-color-duo",
+        "name": "Resistor Color Duo",
+        "uuid": "92387ac7-4c6d-4568-819e-e037b1d86a72",
+        "prerequisites": [
+          "atoms",
+          "lists",
+          "pattern-matching",
+          "maps"
+        ],
+        "practices": [
+          "maps"
+        ],
```

Comment:
Interesting. With those prerequisites, `resistor-color-duo` would be unlocked before `resistor-color` because `resistor-color` has an implementation with a list and `Enum.find_index(@colors, &(&1 == color))`, so it requires `enum`, and `enum` comes after all those prerequisites.

We cannot edit `resistor-color` to NOT use `enum` because it needs to return all the colors in order, and map keys aren't ordered.

I don't know if that's a problem, but it defies my expectations. I would think that `resistor-color` is easier than `resistor-color-duo`, and that `resistor-color-duo` is easier than `resistor-color-trio` 🤔

Reply:
But this one has `maps`, does that come after `lists` and `enum` or is it a different branch?
I could use a list implementation to make them all consistent. To be honest I didn't notice `resistor-color` existed haha.

Reply:
`lists` and `maps` are required by `enum`. 

I think using a list for this exercise would make it less idiomatic. I guess we need to accept it like it is :)

Reply:
> map keys aren't ordered.

100% agree with this statement, it is a common fallacy for students to believe that map keys are ordered just because they appear so in small maps.  

Not sure if this is a problem though.  This is somewhat similar to the "Instruments of Texas" theme that we reuse in the concept exercises as Maud had designed these exercises as teaching exercises. Each one just teaches something different.

Reply:
Someone going through all three exercises in a row is likely to reuse the same implementation though. Either with maps or lists. I probably would have written `resistor-color ` with maps as well since lookup is `O(1)` rather than `O(n)`.